dask/dataframe/tests/test_merge_column_and_index.py

```diff
@@ -244,3 +245,23 @@ def test_merge_column_with_nulls(repartition):
     dask_result = df1_d.merge(df2_d, how="left", left_on="a", right_index=True)
 
     assert_eq(dask_result, pandas_result)
+
+
+@pytest.mark.parametrize("how", ["inner", "left", "right"])
+def test_single_column_join_regression(
+    ddf_left_single, ddf_right_single, on, how, shuffle_method
+):
+    # won't need this if we standardize divisions typing
+    # see https://github.com/dask/dask/issues/8388
+
+    # compute expected
+    expected = ddf_left_single.merge(
+        ddf_right_single, on=on, how=how, shuffle=shuffle_method
+    )
+
+    # compute with internal function
+    result = single_partition_join(ddf_left_single, ddf_right_single, on=on, how=how)
+
+    assert_eq(result, expected)
+    assert result.divisions == expected.divisions
+    assert len(result.__dask_graph__()) < 30
```

Comment:
I'm a little confused by this test. Why do we need to check the graph length, and compare the result of `merge` vs `single_partition_join` (which should be the same code path)? If we just want a regression test that `single_partition_join`'s divisions are always a tuple, maybe we could just add one line checking that to `test_multi.py`'s series of `test_cheap_single_partition_merge_*` functions?

Personally I'd much prefer to not see a regression test (or fix) for this specific issue, but rather to make the setter suggested in https://github.com/dask/dask/issues/8388. That would solve the problem across the board. And it could easily be tested across the board with a couple lines changed in `assert_eq` and `assert_divisions`.

Reply:
Yeah TBH I didn't put much thought into this test, just copied some of the other tests from the file but made sure to do the comparison between calling `single_partition_join` directly versus using `merge`. I verified that these tests would fail without my small change, I think because `merge` might be relying on `merge_indexed_dataframes` instead of `single_partition_join`.

In general, I agree that we should not have tests like this, and was planning to remove this once whatever solution is decided in #8388 is implemented.